pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentIndexCreationDriverImpl.java

```diff
@@ -313,6 +325,36 @@ private void handlePostCreation()
       buildStarTreeV2IfNecessary(segmentOutputDir);
     }
 
+    Set<IndexType> postSegCreationIndexes = IndexService.getInstance().getAllIndexes().stream()
```

Comment:
for a bit more readability, wrap those new changes to a method like `addIndexIfNecessary`

Reply:
Ack